pkg/arvo/tests/sys/hoon/list.hoon

```diff
@@ -0,0 +1,289 @@
+  ==
+::
+::
+++  test-list-find  ^-  tang
+  ;:  weld
+    ::  Checks checks looking for a single item
+    ::
+    %+  expect-eq
+      !>  (some 0)
+      !>  (find (limo ~[1]) (limo ~[1 1 1 2 3]))
+    ::  checks looking for a sub list
+    %+  expect-eq
+      !>  (some 2)
+      !>  (find (limo ~[1 2 3]) (limo ~[1 1 1 2 3]))
+    ::  Checks when the value is not present
+    %+  expect-eq
+      !>  ~
+      !>  (find (limo ~[6]) (limo ~[1 1 1 2 3]))
+  ==
+::
+++  test-list-flop  ^-  tang
+  ;:  weld
+    ::  A list reversed 2x is the same list
+    ::
+    %+  expect-eq
+      !>  (limo ~[1 1 1 2 3])
+      !>  (flop (flop (limo ~[1 1 1 2 3])))
+    :: An empty list reversed is still empty
+    %+  expect-eq
+      !>  ~
+      !>  (flop (limo ~))
+  ==
+::
+++  test-list-gulf  ^-  tang
+  ;:  weld
+    ::  Check for an incresing list
+    %+  expect-eq
+      !>  (limo ~[0 1 2 3])
+      !>  (gulf 0 3)
+  ==
+::
+++  test-list-join  ^-  tang
+  ;:  weld
+    %+  expect-eq
+      !>  (limo ~[2 1 3])
+      !>  (join 1 (limo ~[2 3]))
+  ==
+::
+++  test-list-lent  ^-  tang
+  ;:  weld
+    %+  expect-eq
+      !>  2
+      !>  (lent (limo ~[2 3]))
+    %+  expect-eq
+      !>  0
+      !>  (lent (limo ~))
+  ==
+::
+++  test-list-levy  ^-  tang
+  ;:  weld
+    :: Tail fails condition
+    %+  expect-eq
+      !>  %.n
+      !>  (levy (limo ~[2 3]) |=(x=@ (gth 3 x)))
+    :: Head fails condition
+    %+  expect-eq
+      !>  %.n
+      !>  (levy (limo ~[2 3]) |=([x=@] (lth 2 x)))
+    :: Both pass condition
+    %+  expect-eq
+      !>  %.y
+      !>  (levy (limo ~[2 3]) |=([x=@] (lth x 10)))
+  ==
+::
+++  test-list-lien  ^-  tang
+  ;:  weld
+    :: only Tail fails condition
+    %+  expect-eq
+      !>  %.y
+      !>  (lien (limo ~[2 3]) |=(x=@ (gth 3 x)))
+    :: only head fails condition
+    %+  expect-eq
+      !>  %.y
+      !>  (lien (limo ~[2 3]) |=([x=@] (lth 2 x)))
+    :: Both pass condition
+    %+  expect-eq
+      !>  %.y
+      !>  (lien (limo ~[2 3]) |=([x=@] (lth x 10)))
+      :: Neither
+    %+  expect-eq
+      !>  %.y
+      !>  (lien (limo ~[2 3]) |=([x=@] (gth 10 x)))
+  ==
+::
+++  test-list-murn  ^-  tang
+  ;:  weld
+    %+  expect-eq
+      !>  (limo ~[6 10])
+      !>  
+      %+  murn  (limo ~[2 3 4 5]) 
+          |=  [x=@] 
+          ^-  (unit)
+          ?:  =((mod x 2) 0)
+            [~]
+          (some (mul x 2))
+  ==
+::
+++  test-list-oust  ^-  tang
+  ;:  weld
+    %+  expect-eq
+      !>  (limo ~[2 5])
+      !>  (oust [1 2] (limo ~[2 3 4 5]))
+    %+  expect-eq
+      !>  (limo ~[2 3 4 5])
+      !>  (oust [1 0] (limo ~[2 3 4 5]))
+  ==
+::
+++  test-list-reap  ^-  tang
+  ;:  weld
+    %+  expect-eq
+      !>  (limo ~[1 1 1])
+      !>  (reap 3 1)
+  ==
+::
+++  test-list-reel  ^-  tang
+  ;:  weld
+  :: Use non-associtive operation
+    %+  expect-eq
+      !>  93
+      !>  
+      %+  reel  (limo ~[100 10 5 2]) 
+          |=  [acc=@ x=@] 
+          (sub acc x)
```

Comment:
The wrapper gate is unnecessary:

```
> (reel (limo ~[100 10 5 2]) sub)
93
```

And the expression should be inline with the `!>`.